BEP3.md

```diff
@@ -0,0 +1,156 @@
+
+### Infrastructure Components
+
+*   **New transaction types on Binance Chain**: HTLT and CHLT transactions are used to lock and claim the asset to swap. The details are covered in the following section.
+*   **Swap smart contracts on Ethereum**: the APS contract are used to lock and claim the asset to swap too. The function should be similar to the new transaction types on Binance Chain. The details of interface covered in the following section.
+*   **Deputy process**: an application run and maintained by Owner to facilitate the automatic and continuous swap activity.
+*   **Client tooling**: wallets or other tools that help Clients to monitor the blockchains to complete the claim (and maybe the lock as well).
+
+### Protocol
+
+The protocol is the execution of 
+
+#### Preparation Stage
+
+1. Owner should issue proper number of tokens on Binance Chain as the Pegged 
+2. Owner should create one address on Binance Chain (as OB in the below diagram), and transfer in enough number of tokens for swap
+3. Owner should deploy the APS Ethereum smart contract on Ethereum, and deposit enough number of tokens for swap into a dedicated address (which both should be public announced).
+4. Client should have an address on both Binance Chain (CB) and Ethereum (CE).
+5. For such swap, Owner should publish all their OB and APS contract address, and also the expected minimum time span, MinLockTime.
+
+#### Usage Expectation and Benefits
+
+1. It is expected that each issuer of the pegged token should set up their own Deputy Process, or rely on 3rd party custodian service providers to manage one.
+2. Via the public balance of OB and APS contract, everyone can check the total pegged token swapped, and ensure the total supply stays clear in the same way as if the token is not pegged
+3. Client gets guarantee on their fund safety. They don’t need to worry too much about the availability of the Deputy process, or any cheat from issuers or swap service providers.
+4. It is possible for one set of infrastructure to handle swap for multiple tokens.
+
+#### Client Swap Tokens from Ethereum to Binance Chain
+
+1. Client calls APS contract, with the hash of a secret, X tokens and a time span T parameters, to express his/her interest to swap X tokens. If the parameters are good (e.g. T>MinLockTime, enough tokens to swap), the call transaction is recorded on the blockchain.
+2. Deputy process monitors the events of the APS contract on Ethereum. If it detects the client’s call and verify good, it will sign and broadcast the HTLT transaction on Binance Chain. This will lock X (or more as bonus) number of pegged tokens on Binance Chain. Please note the time span used in this HTLT transaction should be smaller enough than T in Client’s APS call.
```

Comment:
we need to check if the contract has enough deposit

Reply:
There is no need to check. The contract will ensure there has been enough deposit on the contract address.